src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,463 @@
+            event.pageX,
+          );
+          let pointerValue = min + (max - min) * percent;
+          pointerValue = roundValueToClosestStep(pointerValue, step, min);
+
+          const closestValueIndex = getClosestValueIndex(
+            currentValues,
+            pointerValue,
+          );
+          const [minVal, maxVal] = getAllowableThumbRange(closestValueIndex);
+          pointerValue = getBoundedValue(pointerValue, minVal, maxVal);
+          if (pointerValue === currentValues[closestValueIndex]) {
+            return;
+          }
+          const newValues = [...currentValues];
+          newValues[closestValueIndex] = pointerValue;
+          setCurrentValues(newValues);
+          onChange?.(newValues);
+        }
+      },
+      [min, max, step, currentValues, getAllowableThumbRange, onChange],
+    );
+
+    const pointerMoveFunctionRef = React.useRef<
+      (event: PointerEvent) => void
+    >();
+    React.useEffect(() => {
+      const ownerDoc = containerRef.current?.ownerDocument;
+      if (pointerMoveFunctionRef.current && ownerDoc) {
+        ownerDoc.removeEventListener(
+          'pointermove',
+          pointerMoveFunctionRef.current,
+        );
+      }
+      pointerMoveFunctionRef.current = handlePointerMove;
+    }, [handlePointerMove]);
+
+    const pointerUpFunctionRef = React.useRef<(event: PointerEvent) => void>();
+    React.useEffect(() => {
+      const ownerDoc = containerRef.current?.ownerDocument;
+      if (pointerUpFunctionRef.current && ownerDoc) {
+        ownerDoc.removeEventListener('pointerup', pointerUpFunctionRef.current);
+      }
+      pointerUpFunctionRef.current = handlePointerUp;
+    }, [handlePointerUp]);
+
+    React.useEffect(() => {
+      const ownerDoc = containerRef.current?.ownerDocument;
+      ownerDoc?.addEventListener('pointermove', handlePointerMove);
+      ownerDoc?.addEventListener('pointerup', handlePointerUp);
+      return () => {
+        ownerDoc?.removeEventListener('pointermove', handlePointerMove);
+        ownerDoc?.removeEventListener('pointerup', handlePointerUp);
+      };
+    }, [handlePointerMove, handlePointerUp]);
+
+    return (
+      <div
+        className={cx(
+          'iui-slider-component-container',
+          disabled && 'iui-disabled',
+          className,
+        )}
+        {...rest}
+      >
+        {minValueLabel && (
+          <span className='iui-slider-min'>{minValueLabel}</span>
+        )}
+        <div
+          ref={refs}
+          className={cx(
+            'iui-slider-container',
+            undefined !== activeThumbIndex && 'iui-grabbing',
+          )}
+          onPointerDown={handlePointerDownOnSlider}
+        >
+          <div className='iui-slider-rail' />
+          {currentValues.map((thumbValue, index) => {
+            const [minVal, maxVal] = getAllowableThumbRange(index);
+            return (
+              <Thumb
+                key={index}
```

Comment:
Probably shouldn't use index as key.
https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318

Reply:
I don't know if this is a real problem in my situation since the number of thumbs are not changing once the slider is mounted.  Also I don't see that being done in the material-ui slider https://github.com/mui-org/material-ui/blob/444b5a223ccab3423fb14baf5bd48fc273f37514/packages/material-ui-unstyled/src/SliderUnstyled/SliderUnstyled.js#L724

Reply:
I thought the user can control the number/order of thumbs using the `values` array

Reply:
Even though I could not get it to work improperly when using index, I changed it to use a unique count id maintained in a ref.